From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sleigh-InSPECtor <sleighinspector@outlook.com>
Date: Mon, 20 May 2024 15:10:33 +0930
Subject: [PATCH] 6545: AArch32: (Thumb32) BranchWritePC does not clear bit-0

AArch32: (Thumb32) fixed mov & add to pc

* BranchWritePC in thumb mode now clears bit-0
---
 Ghidra/Processors/ARM/data/languages/ARM.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/ARM/data/languages/ARM.sinc b/Ghidra/Processors/ARM/data/languages/ARM.sinc
index a77d7bdf2d..c2a4e64f74 100644
--- a/Ghidra/Processors/ARM/data/languages/ARM.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARM.sinc
@@ -217,7 +217,7 @@ macro SetThumbMode(value) {
 #
 # simple branch, not inter-working
 macro BranchWritePC(addr) {
-   pc = addr;
+	pc = addr & (~0x1);
 }
 
 #
-- 
2.45.1

